flare/mgp/mgp.py

```diff
@@ -111,13 +111,21 @@ def __init__(self,
         self.hyps_mask = None
         self.cutoffs = None
 
+        species_labels = []
+        coded_species = []
         for i, ele in enumerate(unique_species):
             if isinstance(ele, str):
-                self.species_labels.append(ele)
-                self.coded_species.append(element_to_Z(ele))
+                species_labels.append(ele)
+                coded_species.append(element_to_Z(ele))
             elif isinstance(ele, int):
-                self.coded_species.append(ele)
-                self.species_labels.append(Z_to_element(ele))
+                coded_species.append(ele)
+                species_labels.append(Z_to_element(ele))
+            else:
+                print("element type not accepted", ele, type(ele))
+        sort_id = np.argsort(coded_species)
+        for i in sort_id:
+            self.coded_species.append(coded_species[i])
+            self.species_labels.append(species_labels[i])
```

Comment:
Nice!

Reply:
A quick question: Is there anything that we could do here to make this future-proofed for coarse graining? Maybe it's thinking too far ahead, but just so that if a user were to pass in a number outside of {1...118} or a species label outside of the periodic table that everything ends up working okay. I think that this might be too far in advance (and the logic here is easy enough that we could fix it later) but something to keep in mind. This could also matter sooner if someone wants to do FLARE with isotopes, to keep track of distinct species.

Reply:
I guess we can change the element_to_Z part and allow hyps_mask to take any number beyond 118. I can change the ParameterHelper to accommodate that.

But we need to think a bit how to make it consistent. Maybe leave a dictionary in the element_coder module that allows users to injet str-integer pair. And the element_to_Z and Z_to_element function can read from that dictionary.